tests/test_association_task.py

```diff
@@ -75,13 +85,36 @@ def test_run(self):
                 [0]):
             self.assertEqual(test_obj_id, expected_obj_id)
 
+    def test_run_trailed_sources(self):
+        """Test the full task by associating a set of diaSources to
+        existing diaObjects when trailed sources are filtered.
+        """
+
+        assocTask = AssociationTask()
+
+        results = assocTask.run(self.diaSources, self.diaObjects, self.exposure)
+
+        self.assertEqual(results.nUpdatedDiaObjects, len(self.diaObjects) - 3)
+        self.assertEqual(results.nUnassociatedDiaObjects, 3)
+        self.assertEqual(len(results.matchedDiaSources),
+                         len(self.diaObjects) - 3)
+        self.assertEqual(len(results.unAssocDiaSources), 1)
+        for test_obj_id, expected_obj_id in zip(
+                results.matchedDiaSources["diaObjectId"].to_numpy(),
+                [1, 2, 3, 4]):
+            self.assertEqual(test_obj_id, expected_obj_id)
+        for test_obj_id, expected_obj_id in zip(
+                results.unAssocDiaSources["diaObjectId"].to_numpy(),
+                [0]):
+            self.assertEqual(test_obj_id, expected_obj_id)
```

Comment:
Why can't you write these like `self.assertEqual(results.matchedDiaSources["diaObjectId"], [1,2,3,4]`? Or at worst use `np.testing.assert_arrays_equal`? The loops make it hard to follow what is being tested.

Reply:
This results in `ValueError: The truth value of an array with more than one element is ambiguous.
Use a.any() or a.all()` for the arrays, and the original person who wrote the tests I followed likely did it in this way to get around this problem. I looked up ways around this, and this popped up as one of the suggested ways of comparing values in an array for unit testing. The other option seems to be `self.assertTrue(np.array_equal(results.matchedDiaSources["diaObjectId"].values, [1,2], equal_nan=True))`, which I've swapped to since it seems a tad more explicit in what its doing.

Reply:
Use `np.testing.assert_arrays_equal` then, instead. That will give explicit information which values mismatch. These cannot be NaN, so there's no need for NaN-safe comparisons.

Reply:
On that note, would it add clarity if I changed the other unit tests to follow this format?

Reply:
That would definitely be helpful, yes! Please do it on a separate commit, though.

Reply:
Swapped to np.testing and also made a separate commit updating the prior unit tests